### PR TITLE
Allow for login on both AP and via SSO

### PR DESCRIPTION
**Pull request by mratford**

### CODE REVIEW

File: pydbtools/utils.py [hunk 1]
```diff
@@ -157,7 +156,22 @@ def get_user_id_and_table_dir(
...
+        # Remove chunk before last ":"
+        user_id.split(":")[-1]
+        # Remove chunk after first "@"
+        .split("@")[0]
+        # Replace remaining dashes with underscores
+        .replace("-", "_")
+    )
+    # Only use permitted characters for AWS Athena databases
+    unique_db_name = "".join(
+        c for c in unique_db_name if c.isalpha() or c.isdigit() or c == "_"
```
Comment (Thomas-Hirsch) [inline pydbtools/utils.py:173]: always appreciate a non-regexy solution for this :) you could condense it as below (though I suppose that isdigit is technically more restrictive than isnumeric; I suspect the chance is pretty low that it'll actually make a difference but I've been around enough to know when not to tempt fate with mad edge cases)
```suggestion
        c for c in unique_db_name if c.isalnum() or c == "_"
```

**Review by Thomas-Hirsch — CHANGES_REQUESTED** [review]
Looks ok to me. I think it would be useful to add tests with at least one example of each anticipated id format, for the benefit of those working on it in future

**Review by Thomas-Hirsch — APPROVED** [review]
